pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -94,7 +94,7 @@ def _is_on_windows(self) -> bool:
 
     def _has_already_handler(self, signum: Signals) -> bool:
         try:
-            return isinstance(signal.getsignal(signum), FunctionType)
+            return callable(signal.getsignal(signum))
```

Comment:
Seeing test failures here

```python
_________________ test_fault_tolerant_sig_handler[True-False] __________________

register_handler = False, terminate_gracefully = True
tmpdir = local('/tmp/pytest-of-root/pytest-0/test_fault_tolerant_sig_handle0')

    @pytest.mark.parametrize("register_handler", [False, True])
    @pytest.mark.parametrize("terminate_gracefully", [False, True])
    @RunIf(skip_windows=True)
    def test_fault_tolerant_sig_handler(register_handler, terminate_gracefully, tmpdir):
    
        if register_handler:
    
            def handler(*_):
                pass
    
            signal.signal(signal.SIGUSR1, handler)
    
        class TestModel(BoringModel):
            def training_step(self, batch, batch_idx):
                if terminate_gracefully or register_handler:
                    os.kill(os.getpid(), signal.SIGUSR1)
                    sleep(0.1)
                return super().training_step(batch, batch_idx)
    
        model = TestModel()
    
        with mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": str(int(terminate_gracefully))}):
    
            trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, limit_train_batches=2, limit_val_batches=0)
            if terminate_gracefully and not register_handler:
                with pytest.raises(ExitGracefullyException):
>                   trainer.fit(model)
E                   Failed: DID NOT RAISE <class 'pytorch_lightning.utilities.exceptions.ExitGracefullyException'>

/__w/pytorch-lightning/pytorch-lightning/tests/trainer/connectors/test_signal_connector.py:56: Failed
```

Reply:
I don't know why this is failing. Locally all tests in `test_signal_connector.py` are run and are successful. And the initial runs for this pull request were successful, see e.g. https://github.com/PyTorchLightning/pytorch-lightning/actions/runs/1451925254. So it must be related to some other pull request that was merged recently.

Reply:
Hey, I added a few commits here but couldn't solve the issue. But I believe it could be related to the missing teardown I have seen over here #10611. In fact, if I include your change in signal connector, the tests pass there. So I suggest we merge my teardown fix first, then attempt to run tests again here.